app/services/fees/estimate_pay_in_advance_service.rb

```diff
@@ -20,10 +20,10 @@ def call
 
       fees = []
 
-      ActiveRecord::Base.transaction do
+      event.transaction do
```

Comment:
To keep the consistency with other DB transaction, could you use `EventsRecord` instead?

```suggestion
      EventsRecord.transaction do
```